pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,198 @@
+#
+# ABC SMC method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import pints
+import numpy as np
+
+
+class ABCSMC(pints.ABCSampler):
+    r"""
+    Implements the ABC-SMC algorithm as describd in [1].
+
+    In each iteration of the algorithm, the following steps occur:
```

Comment:
Before doing this, please explain in words how it works. This way, you can get across things like the "previous intermediate distribution" (what is that? it just appears in step 1. you haven't told the reader they should have one yet)